packages/router-core/src/router.ts

```diff
@@ -1201,6 +1201,46 @@ export class RouterCore<
           return [parentSearch, {}, searchParamError]
         }
       })()
+      const [preMatchState, strictMatchState, stateError]: [
```

Comment:
this is almost the same as search validation. can we move the identical parts into a reusable function?